src/SignalR/server/SignalR/test/HubConnectionHandlerTests.cs

```diff
@@ -54,6 +54,37 @@ public async Task HubsAreDisposed()
             }
         }
 
+        [Fact]
+        public async Task AsyncDisposablesInHubsAreSupported()
+        {
+            using (StartVerifiableLog())
+            {
+                var trackDispose = new TrackDispose();
+                var serviceProvider = HubConnectionHandlerTestUtils.CreateServiceProvider(s =>
+                {
+                    s.AddScoped<AsyncDisposable>();
+                    s.AddSingleton(trackDispose);
+                },
+                LoggerFactory);
+                var connectionHandler = serviceProvider.GetService<HubConnectionHandler<HubWithAsyncDisposable>>();
+
+                using (var client = new TestClient())
+                {
+                    var connectionHandlerTask = await client.ConnectAsync(connectionHandler);
+
+                    var result = (await client.InvokeAsync(nameof(HubWithAsyncDisposable.Test)).OrTimeout());
+                    Assert.NotNull(result);
+
+                    // kill the connection
+                    client.Dispose();
+
+                    await connectionHandlerTask;
+
+                    Assert.Equal(3, trackDispose.DisposeCount);
```

Comment:
Is this three because the HubConnectionHandler happened to resolve and dispose HubWithAsyncDisposable 3 times? That's MockBehavior.Strict levels of finicky 😆

Reply:
1. OnConnectedAsync
1. HubWithAsyncDisposable.Test
1. OnDisconnectedAsync

Reply:
I know *why* the hub was resolved 3 times, but if we consider using the same hub for OnConnectedAsync and OnDisconnectedAsync is a bug, we should have a test that specifically tests that instead of making other tests unnecessarily strict. Not a big deal though, we'll just have one more test to change if we change this behavior.